mobile/native/src/trade/position/handler.rs

```diff
@@ -62,25 +62,29 @@ pub async fn get_positions() -> Result<Vec<Position>> {
     db::get_positions()
 }
 
+pub fn is_order_acceptable_for_position(
+    position: &Position,
+    submitted_order: &Order,
+) -> Result<bool> {
+    Ok(position.direction == submitted_order.direction.opposite()
+        && position.quantity == submitted_order.quantity)
+}
+
 /// Update the position once an order was submitted
 ///
 /// If the new order submitted is an order that closes the current position, then the position will
 /// be updated to `Closing` state.
 pub fn update_position_after_order_submitted(submitted_order: Order) -> Result<()> {
-    if let Some(position) = db::get_positions()?.first() {
-        // closing the position
-        if position.direction == submitted_order.direction.opposite()
-            && position.quantity == submitted_order.quantity
-        {
-            db::update_position_state(position.contract_symbol, PositionState::Closing)?;
-            let mut position = position.clone();
-            position.position_state = PositionState::Closing;
-            event::publish(&EventInternal::PositionUpdateNotification(position));
-        } else {
-            bail!("Currently not possible to extend or reduce a position, you can only close the position with a counter-order");
-        }
-    }
+    let position = get_current_position()?;
```

Comment:
🛠️ This call should not result in an error if there is no position! - if can happen that we don't have a position - so it should not result in an error here.